### DISCUSSION TITLE
Failed to update fuse version 3.14.1 in open-embedded yocto build environment

### DISCUSSION THREAD

**Discussion by wonchulheo** [libfuse/libfuse#788] [Q&A]
Dear Sir.

In current our open-embedded yocto build system, fuse_2.9.4 is being used.
I'm trying to update fuse version to 3.14.1 from 2.9.4 to avoid OSSVS issues below.
https://nvd.nist.gov/vuln/detail/CVE-2018-10906
https://nvd.nist.gov/vuln/detail/CVE-2019-14860
https://nvd.nist.gov/vuln/detail/CVE-2019-14900

by the way, fuse_3.14.1's build environment seems to be a bit different.
makefile can not be found.

ERROR: fuse-3.14.1-r0 do_install: oe_runmake failed
**make: *** No rule to make target `install'.  Stop.**
ERROR: oe_runmake failed

Does anyone have the fuse_3.14.1.bb file?
or 
Can tell me how to fix the error in the do_install step above?

Below is the fuse_3.14.1.bb recipe file I used.
============================================================

SUMMARY = "Implementation of a fully functional filesystem in a userspace program"
DESCRIPTION = "FUSE (Filesystem in Userspace) is a simple interface for userspace \
               programs to export a virtual filesystem to the Linux kernel. FUSE \
               also aims to provide a secure method for non privileged users to \
               create and mount their own filesystem implementations. \
              "
HOMEPAGE = "http://fuse.sf.net"
SECTION = "libs"

LICENSE = "GPLv2 & LGPLv2"

LIC_FILES_CHKSUM = "file://${COREBASE}/LICENSE;md5=4d92cd373abda3937c2bc47fbc49d690"

SRC_URI = "https://github.com/libfuse/libfuse/archive/refs/tags/fuse-${PV}.tar.gz \
           file://fuse.conf \
"

SRC_URI[md5sum] = "a5fd7866571a5fc08cc8d6dd20ba5e79"
SRC_URI[sha256sum] = "99688d68a753ee97eed51d536b216b8ae5d7fe0236c1a45d439357903bfd4d88"

UPSTREAM_CHECK_URI = "https://github.com/libfuse/libfuse/releases"
UPSTREAM_CHECK_REGEX = "fuse\-(?P<pver>2(\.\d+)+).tar.gz"

inherit autotools pkgconfig update-rc.d systemd

INITSCRIPT_NAME = "fuse"
INITSCRIPT_PARAMS = "start 3 S . stop 20 0 6 ."

SYSTEMD_SERVICE_${PN} = ""

DEPENDS = "gettext-native"

PACKAGES =+ "fuse-utils libulockmgr libulockmgr-dev"

RPROVIDES_${PN}-dbg += "fuse-utils-dbg libulockmgr-dbg"

RRECOMMENDS_${PN}_class-target = "kernel-module-fuse libulockmgr fuse-utils"

FILES_${PN} += "${libdir}/libfuse.so.*"
FILES_${PN}-dev += "${libdir}/libfuse*.la"

FILES_libulockmgr = "${libdir}/libulockmgr.so.*"
FILES_libulockmgr-dev += "${libdir}/libulock*.la"

FILES_fuse-utils = "${bindir} ${base_sbindir}"
DEBIAN_NOAUTONAME_fuse-utils = "1"
DEBIAN_NOAUTONAME_${PN}-dbg = "1"

do_configure_prepend() {
    export MOUNT_FUSE_PATH="${base_sbindir}"
}

do_install_append() {
    rm -rf ${D}/dev

    if ${@bb.utils.contains('DISTRO_FEATURES', 'sysvinit', 'false', 'true', d)}; then
        rm -rf ${D}${sysconfdir}/init.d/
    fi

    if ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'true', 'false', d)}; then
        install -d ${D}${sysconfdir}/modules-load.d
        install -m 0644 ${WORKDIR}/fuse.conf ${D}${sysconfdir}/modules-load.d
    fi
}

do_install_append_class-nativesdk() {
    install -d ${D}${sysconfdir}
    mv ${D}/etc/* ${D}${sysconfdir}/
    rmdir ${D}/etc
}

BBCLASSEXTEND = "native nativesdk"

============================================================

**Comment by Nikratio** [comment]
libfuse does not use Makefiles anymore. To build it, follow the instructions in the README file: https://github.com/libfuse/libfuse#installation
